### DISCUSSION TITLE
Why does PythonEngine.Initialize(); fail and cause the entire C# program to exit?

### DISCUSSION THREAD

**Discussion by Aloento** [pythonnet/pythonnet#1921] [General]
As the title, if I try to set the wrong PATH to initialize Python, I get an error like `ModuleNotFoundError: No module named 'encodings'`, and it causes the whole dotnet to exit.
I would like to be able to handle this error in C#.

**Comment by lostmsu** [comment]
Unfortunately, this is not possible, because C Python does not provide that: https://docs.python.org/3/c-api/init.html#c.Py_Initialize

> There is no return value; it is a fatal error if the initialization fails.

**Reply by Aloento** [reply]
Sad

**Reply by lostmsu** [reply]
@Aloento you can probably work around this by spawning a child process to test the configuration.

**Reply by Aloento** [reply]
yes I do.
https://github.com/Aloento/CraftVits/blob/e102650c71c7961325aba6c490c804fd871ee87d/VitsCLI/Env.cs#L9
